### PR TITLE
fix/password strength indicator ux

### PR DESCRIPTION
**Pull request by bit2world**
#2474

### CODE REVIEW

File: src/app/pages/onboarding/set-password/set-password.tsx [hunk 1]
```diff
@@ -108,7 +108,7 @@ export const SetPasswordPage = () => {
       .defined()
       .when('password', {
         is: (val: string | any[]) => (val && val.length > 0 ? true : false),
-        then: yup.string().oneOf([yup.ref('password')], 'Passwords must match'),
+        then: yup.string().oneOf([yup.ref('password'), null], 'Passwords must match'),
```
Comment (kyranjamie) [inline src/app/pages/onboarding/set-password/set-password.tsx]: What does adding the `null` value here, do?

File: src/app/pages/onboarding/set-password/set-password.tsx [hunk 2]
```diff
@@ -119,7 +119,7 @@ export const SetPasswordPage = () => {
         onSubmit={onSubmit}
         validationSchema={validationSchema}
         validateOnBlur={false}
-        validateOnChange={false}
+        // validateOnChange={false}
```
Comment (kyranjamie) [inline src/app/pages/onboarding/set-password/set-password.tsx]: This should be removed rather than commented out

**Review by kyranjamie — COMMENTED** [review]
Thanks for the contribution 🥳 

Does this also change the text of the mismatching password form 

image

We should make this say something like "Enter password again to confirm"

File: src/app/pages/onboarding/set-password/set-password.tsx (hunk 1 above)
Comment (bit2world) [reply src/app/pages/onboarding/set-password/set-password.tsx]: This include null case, but it maybe not useful in our case

**Comment by bit2world** [comment]
> Thanks for the contribution 🥳
> 
> Does this also change the text of the mismatching password form
> 
> image
> 
> We should make this say something like "Enter password again to confirm"

- Change confirmation Message as `Enter password again to confirm`
- Change the logic of validation
There are several cases of validation in password form
image
- Disable button until validation

**Comment by fbwoolf** [comment]
I am going to cherry-pick these commits and finish this in a new PR. Thanks for the contribution @talentdev0807!

**Pull request closed without merging**